ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/ServerOptionsTest.java

```diff
@@ -57,6 +61,26 @@ public void shouldUseQueryFileParamFromCmdLineInPreferenceToProperties() throws
     final Properties properties = new Properties();
     properties.put(ServerOptions.QUERIES_FILE_CONFIG, "blah");
     assertThat(options.getQueriesFile(properties).get(), equalTo(cmdLineArg));
+  }
+
+  @Test
+  public void shouldOverrideFilePropertiesWithSystemProperties() throws IOException {
+    System.setProperty("bootstrap.servers", "blah:9092");
```

Comment:
Setting & relying on system properties in tests is nasty.  Makes it hard to run tests in parallel, etc.

I know the class under test is all about system properties, but you could just inject a `SystemPropertiesProvide`, which would do away with the new for setting those global props in the test..

Reply:
Yeah - i should at least unset them in this case. I'm not that worried about parallel at the moment. `SystemPropertiesProvider`?

Reply:
Not just parallel. Also later tests passing / failing depending on which prior tests have run, etc, etc.   It's just pain down the line...

Reply:
agreed